src/main/java/it/eng/idsa/businesslogic/processor/consumer/websocket/server/HttpWebSocketServerBean.java

```diff
@@ -0,0 +1,119 @@
+package it.eng.idsa.businesslogic.processor.consumer.websocket.server;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.server.*;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.BindException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Jetty Server instantiation with WebSocket over SSL
+ *
+ * @author Antonio Scatoloni
+ */
+public class HttpWebSocketServerBean {
+    private static final Logger logger = LogManager.getLogger(HttpWebSocketServerBean.class);
+    public static final String WS_URL = "/incoming-data-channel-received-message";
+
+    @Value("${application.idscp.server.port}")
+    private int idscpServerPort;
+
+    @Value("${server.ssl.key-store}")
+    private String keyStoreLocation;
+
+    @Value("${server.ssl.key-password}")
+    private String keyStorePassword;
+
+    private Server server;
+
+    @Autowired
+    private ResourceLoader resourceLoader;
+
+    public Server createServer() {
+        if (null == server || !server.isStarted() || !server.isRunning()) {
```

Comment:
- We should analyze this line of thecode depends of the logic in the other parts in the application...